majifix311/src/main/java/com/example/majifix311/ui/ProblemDetailActivity.java

```diff
@@ -0,0 +1,167 @@
+        intent.putExtra(PROBLEM_INTENT, problem);
+        callingActivity.startActivity(intent);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_problem_detail);
+
+        // this activity requires a problem to be sent in the intent
+        if (getIntent() == null) {
+            finish();
+            return;
+        }
+        mProblem = getIntent().getParcelableExtra(PROBLEM_INTENT);
+
+        setToolbar();
+        setFab();
+
+        mIvMap = (ImageView) findViewById(R.id.iv_map);
+        mIvPhoto = (ExpandableImageView) findViewById(R.id.iv_photo);
+        mTvReporter = (TextView) findViewById(R.id.tv_reporter);
+        mTvCategory = (TextView) findViewById(R.id.tv_category);
+        mTvLocation = (TextView) findViewById(R.id.tv_location);
+        mTvDescription = (TextView) findViewById(R.id.tv_description);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        setProblemData();
+    }
+
+    private void setToolbar() {
+        // disable default animation so ticket number is always at top
+        CollapsingToolbarLayout layout = (CollapsingToolbarLayout) findViewById(R.id.ctl_problemDetail);
+        layout.setTitleEnabled(false);
+
+        // set title to appbar
+        mToolbar = (Toolbar) findViewById(R.id.tb_problemDetail);
+        if (mToolbar != null && mProblem.getTicketNumber() != null) {
+            mToolbar.setTitle(mProblem.getTicketNumber());
+        }
+    }
+
+    private void setFab() {
+        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab_problemDetail);
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                callUser();
+            }
+        });
+    }
+
+    private void setProblemData() {
+        setMap();
+        setExpandableAttachment();
+
+        // set other data to text views
+        if (mProblem.getReporter() != null) {
+            String name = mProblem.getReporter().getName();
+            String phone = mProblem.getReporter().getPhone();
+            String toDisplay = name == null ? phone :
+                    name + (phone == null ? "" : "\n+"+phone);
+
+            if (toDisplay != null) {
+                mTvReporter.setText(toDisplay);
+            }
+        }
+        String category = mProblem.getCategory() == null ? null : mProblem.getCategory().getName();
+        if (category != null) {
+            mTvCategory.setText(category);
+        }
+        if (mProblem.getAddress() != null) {
+            mTvLocation.setText(mProblem.getAddress());
+        }
+        if (mProblem.getDescription() != null) {
+            mTvDescription.setText(mProblem.getDescription());
+        }
+    }
+
+    private void setMap() {
+        // add map only when view is measured (so as to get correct sized map from server)
```

Comment:
This is a new technique for me. I gather the Runnable is enqueued at a point in time where the measure process is already queued/running?

Reply:
That is correct.